sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/common/exception/model/dto/ApiError.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.api.common.exception.model.dto;
+
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+/**
+ * Record for representing the API error.
+ *
+ * @param description  A human-readable description of the error state.
+ * @param code         The internal number referencing the error state.
+ * @param internalTxId The internal tracking number of the request.
+ */
+public record ApiError(
+
+        @Schema(description = "A human readable description of the error state.", example = "Could not create collection `custom-users`, it has invalid characters. Valid characters are alphanumeric and underscores.")
+        String description,
+
+        @Schema(description = "The internal number referencing the error state.", example = "22000")
+        int code,
+
+        // TODO Is this still needed and used? This was not part of the public API.
```

Comment:
Seems generally useful? But agreed, should be documented, public, if so.

Reply:
In the current `restapi` project this is not documented as the API parameter.. I also think it's not part of any examples, this is why I ask..

Reply:
Yeah I think that API endpoints were just copied as-is; but auth token handling changed a few times to/from explicitly passing vs. handled by filter. Or something similar.
But bottomline being that the way restapi does may well be lacking things.